controller/aks-cluster-config-handler.go

```diff
@@ -709,16 +711,30 @@ func (h *Handler) updateUpstreamClusterState(ctx context.Context, secretsCache w
 
 	// check tags for update
 	if config.Spec.Tags != nil {
+
 		if !reflect.DeepEqual(config.Spec.Tags, upstreamSpec.Tags) {
 			logrus.Infof("Updating tags for cluster [%s]", config.Spec.ClusterName)
 			tags := containerservice.TagsObject{
 				Tags: *to.StringMapPtr(config.Spec.Tags),
 			}
-			_, err = resourceClusterClient.UpdateTags(ctx, config.Spec.ResourceGroup, config.Spec.ClusterName, tags)
+			response, err := resourceClusterClient.UpdateTags(ctx, config.Spec.ResourceGroup, config.Spec.ClusterName, tags)
 			if err != nil {
 				return config, err
 			}
-			return h.enqueueUpdate(config)
+			// Azure may have a policy that automatically adds upstream default tags to a cluster resource. We don't
+			// have a good way to detect that policy. We handle this case by checking if Azure returns an unexpected
+			// state for the tags and if so, log the response and move on. Any upstream tags regenerated on the cluster
+			// by Azure will be synced back to rancher.
+			upstreamTags, err := GetDefaultUpstreamTags(response)
+
+			if err != nil || (!reflect.DeepEqual(tags, upstreamTags) && response.Response().StatusCode == 200) {
+				logrus.Infof("Azure API response was not expected. Tags were requested to be updated to [%1s]. " +
+					"Azure returned the updated tags as [%2s]. Completing tag update for cluster [%3s]", to.StringMap(tags.Tags),
```

Comment:
```suggestion
				logrus.Infof("Azure API response was not expected. Tags were requested to be updated to [%1s]. Azure returned the updated tags as [%2s]. Completing tag update for cluster [%3s]", to.StringMap(tags.Tags),
```

Reply:
Also, things like `%1s` means a string formatted with width of `1`. I don't think that is what you want here.